code/game/machinery/spaceheater.dm

```diff
@@ -1,48 +1,69 @@
-#define HEATER_MODE_STANDBY	"standby"
-#define HEATER_MODE_HEAT	"heat"
-#define HEATER_MODE_COOL	"cool"
+#define HEATER_MODE_STANDBY "standby"
+#define HEATER_MODE_HEAT "heat"
+#define HEATER_MODE_COOL "cool"
+#define HEATER_MODE_AUTO "auto"
 
 /obj/machinery/space_heater
 	anchored = FALSE
 	density = TRUE
 	interaction_flags_machine = INTERACT_MACHINE_WIRES_IF_OPEN | INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_OPEN
 	icon = 'icons/obj/atmos.dmi'
 	icon_state = "sheater-off"
+	base_icon_state = "sheater"
 	name = "space heater"
 	desc = "Made by Space Amish using traditional space techniques, this heater/cooler is guaranteed not to set the station on fire. Warranty void if used in engines."
 	max_integrity = 250
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 80, "acid" = 10, "stamina" = 0)
 	circuit = /obj/item/circuitboard/machine/space_heater
+	//We don't use area power, we always use the cell
 	use_power = NO_POWER_USE
-	var/obj/item/stock_parts/cell/cell
+
+	///The cell we spawn with
+	var/obj/item/stock_parts/cell/cell = /obj/item/stock_parts/cell
+	///Is the machine on?
 	var/on = FALSE
+	///What is the mode we are in now?
 	var/mode = HEATER_MODE_STANDBY
-	var/setMode = "auto" // Anything other than "heat" or "cool" is considered auto.
-	var/targetTemperature = T20C
-	var/heatingPower = 20000
+	///Anything other than "heat" or "cool" is considered auto.
+	var/set_mode = HEATER_MODE_AUTO
+	///The temperature we trying to get to
+	var/target_temperature = T20C
+	///How much heat/cold we can deliver
+	var/heating_power = 40000
+	///How efficiently we can deliver that heat/cold (higher indicates less cell consumption)
 	var/efficiency = 20000
-	var/temperatureTolerance = 1
-	var/settableTemperatureMedian = 30 + T0C
-	var/settableTemperatureRange = 30
+	///The amount of degrees above and below the target temperature for us to change mode to heater or cooler
+	var/temperature_tolerance = 1
+	///What's the middle point of our settable temperature (30 °C)
+	var/settable_temperature_median = 30 + T0C
+	///Range of temperatures above and below the median that we can set our target temperature (increase by upgrading the capacitors)
+	var/settable_temperature_range = 30
+	///Should we add an overlay for open spaceheaters
+	var/display_panel = TRUE
+
+	var/maxTemp = 30
+
+	var/minTemp = 30
 
 /obj/machinery/space_heater/get_cell()
 	return cell
 
 /obj/machinery/space_heater/Initialize(mapload)
 	. = ..()
-	cell = new(src)
-	update_icon()
+	if(ispath(cell))
+		cell = new cell(src)
+	update_appearance()
 
-/obj/machinery/space_heater/on_construction()
-	qdel(cell)
-	cell = null
-	panel_open = TRUE
-	update_icon()
-	return ..()
+	maxTemp = (settable_temperature_median + settable_temperature_range) - T0C
```

Comment:
why. What is this, no
calculate ON USE, do not make a global var that is only used for alt clicking.

Replace uses of maxTemp with this equation directly